package com.github.mikephil.charting.buffer;

/* JADX INFO: loaded from: classes.dex */
public class HorizontalBarBuffer extends com.github.mikephil.charting.buffer.BarBuffer {
    public HorizontalBarBuffer(int i, int i2, boolean z) {
        super(i, i2, z);
    }

    /* JADX WARN: Can't rename method to resolve collision */
    /* JADX WARN: Multi-variable type inference failed */
    @Override // com.github.mikephil.charting.buffer.BarBuffer, com.github.mikephil.charting.buffer.AbstractBuffer
    public void feed(com.github.mikephil.charting.interfaces.datasets.IBarDataSet iBarDataSet) {
        float f;
        float fAbs;
        float fAbs2;
        float f2;
        float entryCount = iBarDataSet.getEntryCount() * this.phaseX;
        float f3 = this.mBarWidth / 2.0f;
        for (int i = 0; i < entryCount; i++) {
            com.github.mikephil.charting.data.BarEntry barEntry = (com.github.mikephil.charting.data.BarEntry) iBarDataSet.getEntryForIndex(i);
            if (barEntry != null) {
                float x = barEntry.getX();
                float y = barEntry.getY();
                float[] yVals = barEntry.getYVals();
                if (!this.mContainsStacks || yVals == null) {
                    float f4 = x - f3;
                    float f5 = x + f3;
                    if (this.mInverted) {
                        f = y >= 0.0f ? y : 0.0f;
                        if (y > 0.0f) {
                            y = 0.0f;
                        }
                    } else {
                        float f6 = y >= 0.0f ? y : 0.0f;
                        if (y > 0.0f) {
                            y = 0.0f;
                        }
                        float f7 = y;
                        y = f6;
                        f = f7;
                    }
                    if (y > 0.0f) {
                        y *= this.phaseY;
                    } else {
                        f *= this.phaseY;
                    }
                    addBar(f, f5, y, f4);
                } else {
                    float f8 = -barEntry.getNegativeSum();
                    int i2 = 0;
                    float f9 = 0.0f;
                    while (i2 < yVals.length) {
                        float f10 = yVals[i2];
                        if (f10 >= 0.0f) {
                            fAbs = f10 + f9;
                            fAbs2 = f8;
                            f8 = f9;
                            f9 = fAbs;
                        } else {
                            fAbs = java.lang.Math.abs(f10) + f8;
                            fAbs2 = java.lang.Math.abs(f10) + f8;
                        }
                        float f11 = x - f3;
                        float f12 = x + f3;
                        if (this.mInverted) {
                            f2 = f8 >= fAbs ? f8 : fAbs;
                            if (f8 > fAbs) {
                                f8 = fAbs;
                            }
                        } else {
                            float f13 = f8 >= fAbs ? f8 : fAbs;
                            if (f8 > fAbs) {
                                f8 = fAbs;
                            }
                            float f14 = f8;
                            f8 = f13;
                            f2 = f14;
                        }
                        addBar(f2 * this.phaseY, f12, f8 * this.phaseY, f11);
                        i2++;
                        f8 = fAbs2;
                    }
                }
            }
        }
        reset();
    }
}
